jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -0,0 +1,271 @@
+            'spectrum-viewer').get('id')
+
+        # Subsets are global and are not linked to specific viewer instances,
+        # so it's not required that we match any specific ids for that case.
+        # However, if the msg is not none, check to make sure that it's the
+        # viewer we care about.
+        if msg is not None and msg.viewer_id != self._viewer_id:
+            return
+
+        viewer = self.app.get_viewer('spectrum-viewer')
+
+        self._viewer_data = self.app.get_data_from_viewer('spectrum-viewer')
+
+        self.dc_items = [layer_state.layer.label
+                         for layer_state in viewer.state.layers]
+
+        self.update_ui()
+
+    def update_ui(self):
+        """
+        Set up UI to have all values of currently visible spectra.
+        """
+        if len(self.dc_items) < 1:
+            self.selected_data = ""
+            self.current_flux_unit = ""
+            self.current_spectral_axis_unit = ""
+            return
+
+        self.selected_data = self.app.get_viewer("spectrum-viewer").state.reference_data.label
+
+        self.spectrum = self._viewer_data[self.selected_data]
+
+        # Set UI label to show current flux and spectral axis units.
+        self.current_flux_unit = self.spectrum.flux.unit.to_string()
+        self.current_spectral_axis_unit = self.spectrum.spectral_axis.unit.to_string()
+
+        # Populate drop down with all valid options for unit conversion.
+        self.spectral_axis_unit_equivalencies = self.create_spectral_equivalencies_list()
+        # self.flux_unit_equivalencies = self.create_flux_equivalencies_list()
+        self.flux_unit_equivalencies = self._locally_defined_flux_units()
+
+    def vue_unit_conversion(self, *args, **kwargs):
+        """
+        Runs when the `apply` button is hit. Tries to change units if `new` units are set
+        and are valid.
+        """
+
+        print(self.new_spectral_axis_unit)
+        print(self.new_flux_unit)
+        set_spectral_axis_unit = self.spectrum.spectral_axis
+        set_flux_unit = self.spectrum.flux
+
+        # Try to set new units if set and are valid.
+        if self.new_spectral_axis_unit != "" and\
+                self.new_spectral_axis_unit != self.current_spectral_axis_unit:
+            try:
+                set_spectral_axis_unit = self.spectrum.spectral_axis.to(u.Unit(self.new_spectral_axis_unit))
+            except ValueError:
+                snackbar_message = SnackbarMessage(
+                    f"Unable to convert spectral axis units for selected data. Try different units.",
+                    color="error",
+                    sender=self)
+                self.hub.broadcast(snackbar_message)
+
+                return
+        print(type(set_spectral_axis_unit))
+
+        print(self.spectrum.flux.value)
+        # Try to set new units if set and are valid.
+        if self.new_flux_unit != "" and self.new_flux_unit != self.current_flux_unit:
```

Comment:
I can't currently change _just_ the flux or spectral axis unit; I get a `TypeError: None is not a valid Unit`. Seems like this might also want to check for `None`?

```suggestion
        if self.new_flux_unit is not None and self.new_flux_unit != "" and self.new_flux_unit != self.current_flux_unit:
```